70 login midfi and login flow

margoglvz opened the pull request:
## Description
- Implemented Login and Forgot Password midfi pages, including their routes
- Implemented functionality of the input boxes 

## Screenshots
<img width="1094" alt="Screen Shot 2025-02-16 at 1 51 36 PM" src="https://github.com/user-attachments/assets/6e2e2d00-6e90-4dd8-b14a-dcebcde6b2b7" />
<img width="883" alt="Screen Shot 2025-02-16 at 1 51 48 PM" src="https://github.com/user-attachments/assets/9dd3392e-8d96-4e9b-8d69-26e5686d8ed6" />

<img width="806" alt="Screen Shot 2025-02-16 at 1 52 13 PM" src="https://github.com/user-attachments/assets/692e2ea6-27b7-4905-ab79-887b0654b1df" />

## Issues
Closes #70

jessieh9 reviewed (changes requested):
Hi @margoglvz @HazelYuAhiru , good work so far, just a couple slight changes I'd like:

- [x] For the new route you guys added in users.js, do we think it's a bit redundant? Correct me if I'm misunderstanding, but it seems that it grabs the edit_perms for a user specified by email. In this case why don't we just use the route `/email/:email` which grabs a user by email and you can select the edit_perms from there?

### Signup
- [x] This wasn't mentioned before, but could you guys add when clicking on Cancel button to redirect back to Login page?
- [ ] Can you also add the View/Hide Password for the Password and Confirm password fields here too please?

### Forgot Password
- [ ] Small issue here, but can you edit the routes in App.jsx? I specified a path for it to be linked to
<img width="383" alt="image" src="https://github.com/user-attachments/assets/07559dc7-c143-4d99-8d89-669a53c614fa" />

### Set New Password
- [ ] Similarly, here can you guys add the view/hide password icons
- [ ] Also, if I don't click on anything and just press Enter when the page loads up, it allows me to go into the Set New Password Success page, can you guys take a look into this?
  - I have a feeling it's because the Enter button is just Link to Login so maybe try to have a check there or something.

Other than that, really good work and the frontend matches the Figma really nicely. Thank you guys!

margoglvz commented:
![image](https://github.com/user-attachments/assets/767376eb-8644-4710-b51d-fbd3bacd36ad)
![image](https://github.com/user-attachments/assets/6c787d40-5638-4a69-9da0-feb525907cbd)


- Fixed API endpoint in users.js
- Added Login Redirect on Cancel Button in Signup
- Edited routes
- Implemented View/Hide Password for the Password and Confirm password fields for ForgotPassword and SetNewPassword
- Fixed Enter Requirement on SetNewPassword page

ty! :D

jessieh9 reviewed (approved):
Great work guys, thanks for the fixes!
